tests/test_sampleset.py

```diff
@@ -158,3 +165,16 @@ def test_functional_str(self):
         s = json.dumps(samples.to_serializable())
         new_samples = dimod.SampleSet.from_serializable(json.loads(s))
         self.assertEqual(samples, new_samples)
+
+
+@unittest.skipUnless(_pandas, "no pandas present")
+class TestSampleSet_to_pandas_dataframe(unittest.TestCase):
+    def test_simple(self):
+        samples = dimod.SampleSet.from_samples(([[-1, 1, -1], [-1, -1, 1]], 'abc'),
+                                               dimod.SPIN, energy=[-.5, .5])
+        df = samples.to_pandas_dataframe()
+
+        other = pd.DataFrame([[-1, 1, -1, -.5, 1], [-1, -1, 1, .5, 1]],
+                             columns=['a', 'b', 'c', 'energy', 'num_occurrences'])
+
+        pd.testing.assert_frame_equal(df, other, check_dtype=False)
```

Comment:
Please consider adding a test for duplicate samples. When these occur they will produce duplicate rows in the DF. For example if `sample1 = {'a':1, 'b':0}` is sampled  twice by QC it should show up in the sample set with  `num_occurrences =2`. I find that this is not always true (no clue why), and that I sometimes get two separate instances of `sample1` in a returned Response object. I have resorted to a groupby in pandas to account for this and get the correct `num_occurrences`.

Reply:
I think the best approach would be to add an aggregator to the `SampleSet` object that can be invoked before `to_pandas_dataframe`. See #268